src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ReflectionDomainSetupImplementation.cs

```diff
@@ -36,7 +36,7 @@ public sealed override Exception CreateNonInvokabilityException(MemberInfo perta
                 if (methodBase is ConstructorInfo)
                 {
                     Type declaringType = methodBase.DeclaringType;
-                    if (typeof(Delegate).IsAssignableFrom(declaringType))
+                    if (declaringType.BaseType == typeof(MulticastDelegate))
```

Comment:
Avoids this:

https://github.com/dotnet/runtime/blob/45f5f16cd73b0f6ad5cb153f45cf40fbd2612a11/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/General/Assignability.cs#L39-L46